geoportal/c2cgeoportal_geoportal/scripts/theme2fts.py

```diff
@@ -289,3 +289,53 @@ def _add_layer(
             self._add_fts(layer, interface, "add_layer", role)
 
         return fill
+
+    def _render_label(
+        self,
+        item: "c2cgeoportal_commons.models.main.TreeItem",
+        lang: str,
+    ) -> Optional[str]:
+        patterns = item.get_metadata("searchLabelPattern")
+        if not patterns:
+            return None
+        pattern = patterns[0]
+        tree_paths = list(self._get_paths(item))
+        # Remove paths where the last element isn't a theme
+        tree_paths = [p for p in tree_paths if p[-1].item_type == "theme"]
+        result = None
+        if tree_paths:
+            for path in tree_paths:
+                if len(path) == 2:
+                    current_result = str(pattern.value).format(
+                        name=self._[lang].gettext(item.name),
+                        theme=self._[lang].gettext(path[-1].name),
+                    )
+                elif len(path) > 2:
+                    current_result = str(pattern.value).format(
+                        name=self._[lang].gettext(item.name),
+                        theme=self._[lang].gettext(path[-1].name),
+                        previous_group=self._[lang].gettext(path[1].name),
+                        block=self._[lang].gettext(path[-2].name),
+                    )
+                if result and current_result != result:
+                    sys.stderr.write(
+                        f"Error: the item {item.name} has a label pattern and inconsistent multiple parents\n"
+                    )
+                    sys.exit(1)
+                result = current_result
+                print(current_result)
+            return result
+        return str(pattern.value).format(name=self._[lang].gettext(item.name))
+
+    def _get_paths(
+        self,
+        item: "c2cgeoportal_commons.models.main.TreeItem",
+    ) -> Iterator[List["c2cgeoportal_commons.models.main.TreeItem"]]:
+        if item is None:
+            return
+        if any(item.parents):
+            for parent in item.parents:
+                for path in self._get_paths(parent):
+                    yield [item] + path
```

Comment:
For info, throes are equivalent :-)
```suggestion
                    yield [*item, path]
```

Reply:
Thanks for the suggestion, I put 
`yield [item, *path]`
(I guess that is what you meant)